src/Admin/OnBoarding/Steps/SmsGateway.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace WP_SMS\Admin\OnBoarding\Steps;
+
+use Exception;
+use WP_SMS\Admin\OnBoarding\StepAbstract;
+use WP_SMS\Components\RemoteRequest;
+use WP_SMS\Option;
+
+class SmsGateway extends StepAbstract
+{
+    const CACHE_KEY      = 'wp_sms_gateways';
+    const CACHE_DURATION = 43200; // 12 hours in seconds
+
+    public function getFields()
+    {
+        return ['name'];
+    }
+
+    protected function initialize()
+    {
+        $this->setData('gateways', $this->getAllPages());
+    }
+
+    private function getAllPages()
+    {
+        $cached = get_transient(self::CACHE_KEY);
+        if ($cached !== false) {
+            return $cached;
+        }
+
+        $gateways = array();
+        $page     = 1;
+
+        try {
+            do {
+                $request  = new RemoteRequest('get', "https://staging.wp-sms-pro.com/wp-json/wp/v2/gateway?per_page=100&page={$page}");
+                $response = $request->execute();
+                var_dump($response);
+                if (is_array($response) && !empty($response)) {
+                    $gateways = array_merge($gateways, $response);
+                    $page++;
+                } else {
+                    break;
+                }
+            } while (count($response) === 100);
+
+            // Cache the data
+            set_transient(self::CACHE_KEY, $gateways, self::CACHE_DURATION);
```

Comment:
Let's add a feature to support the cache in RemoteRequest itself like we did in WP Statistics https://github.com/wp-statistics/wp-statistics/blob/master/src/Components/RemoteRequest.php#L109-L161